Vegetarian-Food-Finder/Model/TimeZoneConverter.swift

```diff
@@ -0,0 +1,180 @@
+//
+//  TimeZoneCalculator.swift
+//  Vegetarian-Food-Finder
+//
+//  Created by Abdullahi Ahmed on 7/28/22.
+//
+
+import CoreLocation
+import Foundation
+import UIKit
+
+/// A class that is responsible for date manuplation so the user can make events at their desiered time
+final class TimeZoneConverter {
+    let selectedRestaurant: BusinessDetails
+    init(
+        selectedRestaurant: BusinessDetails
+    ) {
+        self.selectedRestaurant = selectedRestaurant
+    }
+
+    /// computed property that represents the TimeZone of the user
+    private var userCurrnetTime: TimeZone {
+        return TimeZone.current
+    }
+
+    /// Enum that handles the case if the restaurant is closed
+    enum PickedDateError: Error {
+        case invalidDate
+    }
+
+    /**
+      gets the restaurants TimeZone by using reverseGeocoding for locations
+
+     - parameter  completion handler: Gives back the TimeZone of the restaurant
+
+     - Returns: TimeZone of the restaurant
+     */
+    func restaurantLocalTime(
+        completion: @escaping
+        (TimeZone, Error) -> Void
+    ) {
+        let location = CLLocation(
+            latitude: selectedRestaurant.coordinates.latitude,
+            longitude: selectedRestaurant.coordinates.longitude)
+        let geoCoder = CLGeocoder()
+        geoCoder.reverseGeocodeLocation(location) { placeMarks, error in
+            if let placeMark = placeMarks?[0].timeZone, let err = error {
+                completion(placeMark, err)
+            }
+        }
+    }
+
+    /**
+      gets the number of days that restaurant is open
+
+     - Returns: an array of int
+     */
+    func getDayNumber(
+    ) -> [Int] {
+        var pickedDay = [0]
+        let restaurantWorkingHoursDetails = selectedRestaurant.hours.weekDetails
+        for dayNumber in restaurantWorkingHoursDetails {
+            pickedDay.append(dayNumber.day)
+        }
+        return pickedDay
+    }
+
+    /**
+      Checks if the restaurant is open at the date the user picked
+
+     - parameter  pickedDate: UIDatePicker representing the date the user picked
+
+     returns dictionary of Bool(represents If restaurant open) and Int(day that it's open)
+     */
+    func checkIfOpenAt(
+        pickedDate: UIDatePicker
```

Comment:
You don't need to pass the picker, it is better option to store the date when the user changes it and then, pass the date here